torch/utils/viz/MemoryViz.js

```diff
@@ -1402,17 +1402,42 @@ function unpickle(buffer) {
       case LONG1:
         {
           const s = bytebuffer[offset++];
-          if (s > 8) {
-            throw new Error(`Unsupported number bigger than 8 bytes ${s}`);
-          }
-          for (let i = 0; i < s; i++) {
-            scratch_bytes[i] = bytebuffer[offset++];
-          }
-          const fill = scratch_bytes[s - 1] >= 128 ? 0xff : 0x0;
-          for (let i = s; i < 8; i++) {
-            scratch_bytes[i] = fill;
+          if (s <= 8) {
+            for (let i = 0; i < s; i++) {
+              scratch_bytes[i] = bytebuffer[offset++];
+            }
+            const fill = scratch_bytes[s - 1] >= 128 ? 0xff : 0x0;
+            for (let i = s; i < 8; i++) {
+              scratch_bytes[i] = fill;
+            }
+            stack.push(Number(big[0]));
+          } else { // BigInt
+            let scratch_bytes_unbounded = [];
+            for (let i = 0; i < s; i++) {
+              scratch_bytes_unbounded.push(bytebuffer[offset++]);
+            }
+
+            // BigInt can only convert from unsigned hex, thus we need to
+            // convert from twos-complement if negative
+            const negative = scratch_bytes_unbounded[s - 1] >= 128;
```

Comment:
Can this be done using bitwise operators on BigInt to make it faster and shorter? E.g. ``-(~BigInt(`{0x${hex_str}`) + 1)``

Reply:
I don't think so. `~BigInt("0xffffffffffffffff")` is `-18446744073709551616n`, not `0` as it would have to be for `-(~n + 1)` to work as a conversion. I believe that only works with fixed-size twos complement numbers.